### DISCUSSION TITLE
Is custom server supported with app router? Documentation is missing.

### DISCUSSION THREAD

**Discussion by shaunwarman** [vercel/next.js#55283] [App Router] [answered]
For the pages router, the custom server documentation can be found below.
https://nextjs.org/docs/pages/building-your-application/configuring/custom-server

There is no custom server documentation for app router currently. Is that by design? Does that mean custom server not supported with app router going forward?

**Comment by shaunwarman** [comment]
cc @leerob @timneutkens @ijjk

**Comment by danieljee** [comment]
Everything is broken in app router and I really need a custom server to bring back my sanity.

**Comment by leerob** [comment]
Edit: see https://github.com/vercel/next.js/discussions/55283#discussioncomment-10151603

**Reply by shaunwarman** [reply]
Perfect, thanks @leerob !

**Comment by harrytang** [comment]
Hi @leerob , is the documentation for this ready yet ? Thank you!

**Reply by arian-fallahpour** [reply]
I haven't seen it yet

**Reply by xbdirisxk** [reply]
5 months past and still waiting...

**Comment by nont65** [comment]
How to use with typescript?

**Comment by chiefdeals** [comment]
> Yes, it is supported. We will get that page added in both Pages and App Router documentation.

9 months to birth this baby, ya? 😅

**Comment by rpratama-codes** [comment]
still waiting

**Reply by suryavirkapur** [reply]
@leerob any eta?

**Reply by rangga-synergy** [reply]
waiting...

**Comment by massalinux** [comment]
any news on this?

**Comment by coolRoger** [comment]
waiting for it, i need to setup a websocket by uisng custom server

**Comment by leerob** [comment, marked as answer]
Yes, using a custom server is supported.

PR with an example and documentation here: https://github.com/vercel/next.js/pull/68167
